unlock-app/src/components/interface/checkout/main/Metadata.tsx

```diff
@@ -111,11 +111,8 @@ export const MetadataInputs = ({
       'border-brand-secondary hover:border-brand-secondary focus:border-brand-secondary focus:ring-brand-secondary'
   )
 
-  let recipient = account
-  // The first recipient could be hardcoded
-  if (id == 0 && paywallConfig.recipient) {
-    recipient = paywallConfig.recipient
-  }
+  const rec = checkConfig(paywallConfig.recipient, locksConfig.recipient)
```

Comment:
```suggestion
  const rec = recipientFromConfig(paywallConfig.recipient, locksConfig.recipient)
```

Reply:
you can just pass the paywallconfig object and the lock address there so you look up `paywallConfig.locks[lock!.address]` _inside_ the helper function.